packages/amplify-cli/src/__tests__/extensions/amplify-helpers/resource-status.test.ts

```diff
@@ -485,9 +485,11 @@ describe('resource-status', () => {
       const fsMock = fs as jest.Mocked<typeof fs>;
       fsMock.existsSync.mockReturnValue(true);
       const hashElementMock = hashElement as jest.MockedFunction<typeof hashElement>;
+      // eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
these are false-positives in this file.
mocked function has non-promisified overloads that take callback and linter (or tsc?) is matching wrong one here.

Reply:
Nit: maybe we could use `hashElementMock.mockResolvedValue({hash: sampleHash1})` instead? Not sure if that would fix the linter issue